sounds/views.py

```diff
@@ -316,7 +318,9 @@ def sound_download(request, username, sound_id):
                                                     reverse("sound", args=[username, sound_id])))
     sound = get_object_or_404(Sound, id=sound_id, moderation_state="OK", processing_state="OK")
     if sound.user.username.lower() != username.lower():
-        raise Http404
+        user = get_user_from_oldusername(username)
+        if user == None or user.id != sound.user_id:
+            raise Http404
```

Comment:
I think we also need a util function here. Wouldn't it make sense to have a `get_sound_or_404`  function that checks for old usernames as well? This possibly applies to other places as well.